provider/pkg/resources/extension_resource.go

```
@@ -49,6 +58,14 @@ func (r *extensionResource) Check(ctx context.Context, urn resource.URN, inputs,
 		return nil, failures, nil
 	}
 
+	if typedInputs.AutoNaming != nil {
+		autonaming.ApplyAutoNaming(&metadata.AutoNamingSpec{
```

should we return the error if this fails?